### ISSUE TITLE
[Feature request] Methods return self.__class__(...) instead of Version(...)

### ISSUE THREAD

**Issue by dwreeves** [rbarrois/python-semanticversion#112]
I have a use case where I am subclassing `Version`.

Some of the instance methods inside the Version class return a `Version(...)` instead of `self.__class__(...)` (or `cls(...)` in a few classmethods), which makes it a bit more annoying to subclass. It means I basically need to wrap all the methods I need so that they coerce any `Version` output of a `Version` into my own class.

Would it be possible to move toward the other convention that supports subclassing a bit better?

**Comment by rbarrois** [comment]
Hi! That's an interesting feature request :)

I've got to think through the impact of that change; out of curiosity, what kind of extra functionality do you had to your subclass?

**Comment by dwreeves** [comment]
I did two things:

1. Creating a SematicVersion [Pydantic](https://github.com/samuelcolvin/pydantic) model.
2. I need the string representation for `1.2.3-alpha` to be something like `v1_2_3_alpha` because the place where these semantic versions are going doesn't allow for `.`, `-`, and `+` in the strings, and first char needs to be a letter.

Ultimately what I ended up doing was not subclassing (ended up being tricky) and took more of a "composition over inheritance" approach by setting my class's methods to equal the ones I needed from `semantic_version.Version` (stuff like `__ge__` and `next_minor`). The ones that returned `Version` objects I wrapped to return my custom class instead.

My final coded ended up looking something like this:

```python

class MyCustomVersion:

    __ge__ = Version.__ge__
    next_major = _return_my_custom_version(Version.next_major)
```

So in a world in which the methods return `self.__class__()`, I wouldn't have needed to use `_return_my_custom_version`, but ultimately I still probably wouldn't have ended up subclassing in my particular use case for reasons.
